### DISCUSSION TITLE
Improving wiki integration to further promote usage (for beginners)

### DISCUSSION THREAD

**Discussion by Joppe27** [ppy/osu#13325] [Ideas]
Starting this discussion to group ideas about increasing the wiki usage and making it an acutal part of the game that people use more than once every year. The wiki could be very useful to beginners, but as it is they hardly know it exists or can't be bothered to search far enough (understandably, sometimes).

Some proposals (I will be adding more as time goes on):

<details>
  <summary>1. "Help" buttons in settings that link to the wiki</summary>
  
## Explanation
A help button on every big section in the settings. Clicking it opens the 'settings' wiki page and scrolls down to the relevant section.

Players can now easily find out what a setting does and which settings to change to get their desired result (although the settings wiki page isn't the most helpful, but that's not relevant here).

## Example
![settings](https://user-images.githubusercontent.com/56885706/120693341-2db0a300-c4a9-11eb-8834-5cc531786ced.png)

</details>

2. Probably a deticated button ("tech issues? click here!") to the 'help centre' article because a lot of players don't know the article exists, or they think it does not cover _tech_ issues. The article is also very long which makes people click away rather quickly if they're not sure whether it covers their issue or not.

<details>
  <summary>3. Direct hotkey to wiki about an unfamiliar part of osu! without needing to search</summary>
  
## Explanation
A hotkey (F7?) that immediatly brings up the wiki article for an area where the player currently is. 

I feel like this is pretty necessary in the editor, since that part of the client does not speak for itself as much. There is an abundance of wiki articles covering the editor available that would make editing beatmaps so much more accessible if players were able to find them without too much effort. Example: timing screen, "What does xyz do?", press F7, read article, done.

This is especially useful to beginners, as the "I have no idea what I'm doing" levels can get quite high to those who aren't familiar with the client and the terms needed to navigate the gigantic wiki properly (also not a lot of people seem to know about the existence of the osu! wiki anyway). It also allows players to discover all the possibilities of a specific area of the client (e.g. searchbox filtering in song select) without having to dig up 5 year old reddit posts. All of the information is now available in one place and with the press of a button.

Screens/areas that have a link to the wiki can be indicated like the example below, but it would probably be best to only do that the first few times a beginner visits that screen to reduce distraction. As I am typing this I realize that this is probably a little too much to ask so feel free to give suggestions or disregard (I am aware that implementing this is going to be way too much work, just wanted to get the idea out there I guess).

## Example
Pressing F7 would obviously bring up the 'Song Selection' article in this case.

https://user-images.githubusercontent.com/56885706/120693351-31442a00-c4a9-11eb-912d-1a6b880dd9e7.mp4

</details>

Please comment on this post if you think you've got a good idea for the wiki implementation in osu! too, so staff will be able to cherry pick the best ones once the time eventually (ever?) comes where there aren't any other issues left that are more important than this.

**Comment by NafeeJ** [comment]
I think you would have to make it very obvious that you can press F7 (or whatever button) for help but that sounds like a good idea that prevents any issue of trying to integrate a new help button into the UI at random places of the game.

I remember one of the most confusing parts of the game was score/pp so perhaps at the score screen pressing the button would give user choice of going to score or pp wiki.

Edit: Maybe even have an in-game webview for the wiki somewhere?

**Reply by gagahpangeran** [reply]
Wiki overlay is already there.

**Reply by NafeeJ** [reply]
oh cool never noticed that
